### PR TITLE
include/pthread.h: fix PTHREAD_COND_INITIALIZER macro and explanation comment for tinyara/semaphore.h

### PR DESCRIPTION
**Pull request by jeongchanKim**
PTHREAD_COND_INITIALIZER will initialize the pthread_cond_s,and pthread_cond_s is made from semaphore
Referenced by Nuttx(35a6ef7)

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: os/include/pthread.h [hunk 1]
```diff
@@ -284,7 +284,7 @@ struct pthread_cond_s {
 	sem_t sem;
 };
 typedef struct pthread_cond_s pthread_cond_t;
-#define PTHREAD_COND_INITIALIZER { {0, 0xffff} }
+#define PTHREAD_COND_INITIALIZER { SEM_INITIALIZER(0) }
```
Comment (sunghan-chang) [inline os/include/pthread.h]: Let's modify line 80 as well.
```
#include <tinyara/semaphore.h>	/* For sem_t and SEM_* defines */
```

Comment (jeongchanKim) [reply os/include/pthread.h]: done

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**